### PR TITLE
cluster: back store set by cluster peer information

### PR DESCRIPTION
**Pull request by fabxc**
Had a bit of trouble deciding where which code should live. Ultimately decided to move `StoreSet` to `pkg/cluster`. It barely has anything cluster specific yet, but I expect this to change down the line.

### CODE REVIEW

File: pkg/cluster/stores.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	return s.labels
+}
+
+func (s *storeInfo) setLabels(lset []storepb.Label) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+	s.labels = lset
+}
+
+// repeatUntil executes f every interval seconds until stopc is closed.
```
Comment (Bplotka) [inline pkg/cluster/stores.go]: I know it is only 13 lines of code.. but can we have utils package for these small things?

**Comment by Bplotka** [comment]
Having that in pkg/cluster is fine. Even if it related to store nodes the main purpose of it is update the data from peer nodes.

File: scripts/polyglot.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+${POLYGLOT_BIN}  \
+  --command=call \
+  --full_method=promlts.Store/Series \
+  --endpoint="[::]:19090" \
+  --proto_discovery_root=./pkg/store/storepb \
+  --add_protoc_includes=/Users/fabxc/repos/src/github.com/gogo/protobuf/gogoproto,/Users/fabxc/repos/src/github.com/gogo/protobuf,/Users/fabxc/repos/src/github.com/gogo/protobuf/protobuf \
```
Comment (Bplotka) [inline scripts/polyglot.sh]: Worth to add example request somewhere

File: scripts/quickstart.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
```
Comment (Bplotka) [inline scripts/quickstart.sh]: Do we need that now?

File: pkg/cluster/stores.go (hunk 1 above)
Comment (fabxc) [reply pkg/cluster/stores.go]: name suggestions? `pkg/runutil`?

File: test/e2e/query_test.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		if err := cmd.Start(); err != nil {
+			close()
+			return func() {}
+		}
+		closers = append(closers, cmd)
+	}
+	return close
+}
+
+// TestQuerySimple runs the standard quickstart script and verifies that we eventually can read
```
Comment (Bplotka) [inline test/e2e/query_test.go]: Comment is lying (:

File: pkg/cluster/stores.go (hunk 1 above)
Comment (Bplotka) [reply pkg/cluster/stores.go]: just `pkg/util`?

File: scripts/polyglot.sh (hunk 2 above)
Comment (fabxc) [reply scripts/polyglot.sh]: Ops, that was actually not supposed to be included at all :)

File: scripts/quickstart.sh (hunk 3 above)
Comment (fabxc) [reply scripts/quickstart.sh]: I think it's still nice to have for a quick spinup for manual testing.

File: pkg/cluster/stores.go (hunk 1 above)
Comment (fabxc) [reply pkg/cluster/stores.go]: The problem with package names like `common` and `util` is that it doesn't tell you anything about what's in it. It's not good Go practice and has been discussed here: https://blog.golang.org/package-names


Personally I tend towards not extracting small functions like this unless they are really reused often. But for this one it's probably worth doing it.

**Comment by Bplotka** [comment]
https://github.com/improbable-eng/thanos/issues/6

File: scripts/quickstart.sh (hunk 3 above)
Comment (Bplotka) [reply scripts/quickstart.sh]: ok

File: pkg/cluster/stores.go (hunk 1 above)
Comment (Bplotka) [reply pkg/cluster/stores.go]: I know, usually utils and common are just a trash bin for everything that was randomly found and now one has idea where to put it (:

what about `pkg/common/repeat` dir & `repeat` package name? `repeat.Until` sounds like a nice name for this function

Comment (fabxc) [reply pkg/cluster/stores.go]: `common` repo in Prometheus is sad legacy that we'll probably never get rid of :)

If we name the package `repeat`, this function will probably be the only one ever in there. `runutil` could contain various of those small helper functions that care about running things in different ways.

Currently have it locally as `runutil.Repeat(...)`

Comment (Bplotka) [reply pkg/cluster/stores.go]: +1

**Pull request merged**